src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+import React, {
+  useState,
+  useMemo,
+  useCallback,
+  Dispatch,
+} from 'react';
+import { peopleFromServer } from '../data/people';
+import { Person } from '../types/Person';
+
+type Props = {
+  delay: number;
+  onSelected: (username: Person) => void;
+};
+
+const normalize = (value: string) => value.toLowerCase();
+
+const debounce = (callback: Dispatch<string>, delay: number) => {
+  let timerId: number;
+
+  return (...args: string[]) => {
+    clearTimeout(timerId);
+
+    timerId = setTimeout(callback, delay, ...args);
+  };
+};
+
+export const Autocomplete: React.FC<Props> = ({ delay, onSelected }) => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [listIsShown, setListIsShown] = useState('false');
```

Comment:
```suggestion
  const [listIsShown, setListIsShown] = useState(false);
```
the reason list is shown everytime - use real boolean instead of string (fix everywhere)

Reply:
Hi Taras. I have some problems. I fixed the logical problem you wrote about. Yes, I don’t have a list with users  at the beginning with an empty input now. But how do i get rid of the list after i have entered the text and cleared the input.  I tried to fix it but it doesn't work                                                                                                       
    const handleOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { value } = event.target;

    setListIsShown('false');
    setQuery(value);
    applyQuery(value);

    if (value.trim() === '') {
      setListIsShown(false);
    } else {
      applyQuery(value);
      setListIsShown(true);
    }
  };

Reply:
And I read the material on interactive elements, implemented it in my code, but for some reason es lint still gives an error                                                                                                                                                                                              
                <p
                    // role="button"
                    // tabIndex={0}
                    className="has-text-link"
                    onClick={() => handleOnClick(user)}
                  >